sdk/compute/Azure.ResourceManager.Compute/tests/Scenario/GalleryImageVersionContainerTests.cs

```diff
@@ -0,0 +1,216 @@
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task Get()
+        {
+            _resourceGroup = await CreateResourceGroupAsync();
+            var container = await GetGalleryImageVersionContainerAsync();
+            var vmContainer = _resourceGroup.GetVirtualMachines();
+            var vmName = Recording.GenerateAssetName("testVM-");
+            var vnet = await CreateVirtualNetwork();
+            var nic = await CreateNetworkInterface(GetSubnetId(vnet));
+            var vmInput = ResourceDataHelper.GetBasicLinuxVirtualMachineData(DefaultLocation, vmName, nic.Id);
+            var lroVm = await vmContainer.CreateOrUpdateAsync(vmName, vmInput);
+            _vm = lroVm.Value;
+            await _vm.DeallocateAsync();
+            await _vm.GeneralizeAsync();
+            var vmID = _vm.Id;
+            var BasicGalleryImageVersionData = ResourceDataHelper.GetBasicGalleryImageVersionData(DefaultLocation, vmID);
+            var GalleryImageVersionName = "1.0.0";
+            var lroImageVersion = await container.CreateOrUpdateAsync(GalleryImageVersionName, BasicGalleryImageVersionData);
+            GalleryImageVersion imageversion = lroImageVersion.Value;
+            GalleryImageVersion imageversion2 = await container.GetAsync(GalleryImageVersionName);
+
+            ResourceDataHelper.AssertGalleryImageVersion(imageversion.Data, imageversion2.Data);
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task GetAll()
+        {
+            _resourceGroup = await CreateResourceGroupAsync();
+            var container = await GetGalleryImageVersionContainerAsync();
+            var vmContainer = _resourceGroup.GetVirtualMachines();
+            var vmName = Recording.GenerateAssetName("testVM-");
+            var vnet = await CreateVirtualNetwork();
+            var nic = await CreateNetworkInterface(GetSubnetId(vnet));
+            var vmInput = ResourceDataHelper.GetBasicLinuxVirtualMachineData(DefaultLocation, vmName, nic.Id);
+            var lroVm = await vmContainer.CreateOrUpdateAsync(vmName, vmInput);
+            _vm = lroVm.Value;
+            await _vm.DeallocateAsync();
+            await _vm.GeneralizeAsync();
+            var vmID = _vm.Id;
+            var BasicGalleryImageVersionData = ResourceDataHelper.GetBasicGalleryImageVersionData(DefaultLocation, vmID);
+            var GalleryImageVersionName1 = "1.0.0";
+            var GalleryImageVersionName2 = "1.0.0";
+            var input1 = BasicGalleryImageVersionData;
+            var input2 = BasicGalleryImageVersionData;
+            _ = await container.CreateOrUpdateAsync(GalleryImageVersionName1, input1);
+            _ = await container.CreateOrUpdateAsync(GalleryImageVersionName2, input2);
+            int count = 0;
+            await foreach (var galleryImageVersion in container.GetAllAsync())
+            {
+                count++;
+            }
+            Assert.GreaterOrEqual(count, 1);
```

Comment:
We should expect two resource instances on Azure since we invoked the create function twice
```suggestion
            Assert.GreaterOrEqual(count, 2);
```